src/main/daml/Daml/Finance/Instrument/Bond/Util.daml

```diff
@@ -19,41 +19,21 @@ import Daml.Finance.Interface.Lifecycle.Event qualified as Event (I, getEventTim
 import Daml.Finance.Interface.Lifecycle.Observable qualified as Observable (I)
 import Daml.Finance.Interface.Lifecycle.Rule.Lifecycle qualified as Lifecycle (I)
 import Daml.Finance.Interface.Types.Common (Id)
-import Daml.Finance.Interface.Types.Date.Calendar
 import Daml.Finance.Interface.Types.Date.Classes (toUTCTime)
-import Daml.Finance.Interface.Types.Date.DayCount
-import Daml.Finance.Interface.Types.Date.RollConvention
-import Daml.Finance.Interface.Types.Date.Schedule
+import Daml.Finance.Interface.Types.Date.DayCount (DayCountConventionEnum)
+import Daml.Finance.Interface.Types.Date.Schedule (PeriodicSchedule(..), Schedule, SchedulePeriod)
 import Daml.Finance.Interface.Util.Disclosure qualified as Disclosure (I)
 import Daml.Finance.Lifecycle.Effect (Effect(..))
-import Daml.Finance.RefData.HolidayCalendar
+import Daml.Finance.RefData.HolidayCalendar (GetCalendar(..), HolidayCalendar, HolidayCalendarKey(..))
 import Daml.Finance.RefData.Time.DateClock (Unit(..))
-import Daml.Finance.Util.Date.DayCount
-import Daml.Finance.Util.Date.Schedule
+import Daml.Finance.Util.Date.DayCount (calcPeriodDcf)
+import Daml.Finance.Util.Date.Schedule (createSchedule)
 import Prelude hiding (key)
 
 -- | Retrieve holiday calendar(s) from the ledger and create a coupon schedule
-createCouponScheduleAndFrequency : Date -> [Text] -> BusinessDayConventionEnum -> PeriodEnum -> Int -> Date -> Date -> Party -> Party -> Update (Schedule, Frequency)
-createCouponScheduleAndFrequency firstCouponDate holidayCalendarIds businessDayConvention couponPeriod couponPeriodMultiplier issueDate maturityDate issuer calendarDataAgency = do
+createCouponSchedule : PeriodicSchedule -> [Text] -> Party -> Party -> Update Schedule
```

Comment:
What about renaming this function to `rollCouponSchedule`? The schedule has already been created at this stage.

Could you please add a `.` to the doc comment?